lib/spack/llnl/string.py

```diff
@@ -41,6 +41,20 @@ def comma_and(sequence: List[str]) -> str:
     return comma_list(sequence, "and")
 
 
+def ordinal(number: int) -> str:
+    """Return the ordinal representation (1st, 2nd, 3rd, etc.) for the provided number.
+
+    Args:
+        number: int to convert to ordinal number
+
+    Returns: number's corresponding ordinal
+    """
+    idx = (number % 10) << 1
+    tens = number % 100 // 10
+    suffix = "th" if tens == 1 else "thstndrdththththththth"[idx : idx + 2]
```

Comment:
I think this has one unreachable `th` on the end.

Reply:
might be clearer for this reason as:

```suggestion
    idx = (number % 10) << 1
    tens = number % 100 // 10
    suffix = "th" if (tens == 1 or idx > 3) else "thstndrd"[idx : idx + 2]
```